go-controller/pkg/ovn/policy.go

```diff
@@ -551,28 +551,43 @@ func (oc *Controller) localPodAddDefaultDeny(nsInfo *namespaceInfo,
 	if (len(policy.Spec.PolicyTypes) == 1 && policy.Spec.PolicyTypes[0] == knet.PolicyTypeEgress) ||
 		len(policy.Spec.Egress) > 0 || len(policy.Spec.PolicyTypes) == 2 {
 		if oc.lspEgressDenyCache[portInfo.name] == 0 {
-			if err := addToPortGroup(oc.ovnNBClient, nsInfo.portGroupEgressDenyName, portInfo); err != nil {
-				klog.Warningf("Failed to add port %s to egress deny ACL: %v", portInfo.name, err)
-			}
+			addEgressDeny = true
 		}
 		oc.lspEgressDenyCache[portInfo.name]++
 	}
+
+	// we're done with the lsp cache - release the lock before transacting
+	oc.lspMutex.Unlock()
```

Comment:
Is there a chance we could get another `localPodAddDefaultDeny` event while an `addToPortGroup` command is still running? And if so what would happen in that case since the cache would be unlocked/current, but OVN wouldn't necessarily be in the right state?

Reply:
No, because we never work the same pod in parallel. Good question, though!

Reply:
Cool that makes sense, probably another perf bounding point there right? (i.e something we should address once Dave's work makes it in)

Reply:
The standard is for k8s controllers to never process the same object in parallel; that prevents all kinds of otherworldly race conditions, FYI.